src/components/SignUpForm/index.jsx

```diff
@@ -33,13 +33,25 @@ const schema = Joi.object({
 });
 
 export default function SignUpForm({ submit }) {
-  const { register, handleSubmit, errors, reset } = useForm({
+  const { register, handleSubmit, errors, reset, setError } = useForm({
     resolver: joiResolver(schema),
   });
 
-  const onSubmit = (data) => {
-    submit(data);
-    reset();
+  const onSubmit = async (data) => {
+    const status = await submit(data);
+    switch (status) {
+      case "auth/email-already-in-use":
+        setError("email", {
+          type: "already taken",
+          message:
+            "Email is already taken! Please enter another one or sign in.",
```

Comment:
please add translations everywhere

Reply:
I will work on it after finishing these authentication functionality

Reply:
I already wrote you in another PR that I think it will be faster if you add translation now, since we don't have much time left. It will be more time-consuming for us later to add translations, while we have still many other changes.

Reply:
yes